### PR TITLE
portable shared util

### PR DESCRIPTION
**Pull request by ssipasseuth**
This add a standard compliant and more convenient way to manage assets and templates within a pci.
Two util libs are added as shared libs. They rely on the pci markups and custom `<script>` tag  to store additional asset and templates data.
This is best alternative I have found until we have a full-featured PCI SDK.

### CODE REVIEW

**Comment by krampstudio** [comment]
To run shared libraries from the cli or jenkins, you need to define them manually in the test config. For example in your test.html : 

```
         require(['/tao/ClientConfig/config'], function(){
                require.config({
                    config : {
    
                    paths : {
                        'IMSGlobal/jquery_2_1_1':   '../../../taoQtiItem/views/js/portableSharedLibraries/IMSGlobal/jquery_2_1_1',
                        'OAT/lodash':               '../../../taoQtiItem/views/js/portableSharedLibraries/OAT/lodash',
                        'OAT/async':                '../../../taoQtiItem/views/js/portableSharedLibraries/OAT/async',
                        'qtiInfoControlContext':    '../../../taoQtiItem/views/js/runtime/qtiInfoControlContext',
                    }
                });
```

**Comment by ssipasseuth** [comment]
Right ! Sorry, always forgot about that. I will do it now.

**Comment by ssipasseuth** [comment]
ok, unit tests fixed

File: install/scripts/portableSharedLibraries/OAT/util/tpl.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    return function tpl($container){
+        
+        var templates = loadTemplates($container);
+        
+        return {
+            exists : function exists(templateId){
+                return (templateId && templates[templateId]);
+            },
+            render : function render(templateId, data){
+                if(templateId && templates[templateId]){
```
Comment (krampstudio) [inline install/scripts/portableSharedLibraries/OAT/util/tpl.js]: you can reuse your own methods `if(templateId && templates[templateId]){` -> `if(this.exists(templateId`

File: install/scripts/portableSharedLibraries/OAT/util/asset.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        
+        return {
+            exists : function exists(id){
+                return (id && assets[id]);
+            },
+            get : function get(id){
+                return assets[id] || '';
+            },
+            getAll : function(){
+                return _.clone(assets);
```
Comment (krampstudio) [inline install/scripts/portableSharedLibraries/OAT/util/asset.js]: Is it necessary to clone the assets ?

**Pull request merged**